### PR TITLE
Bug 1777954 - Signature lists overhaul

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

**Comment by willkg** [comment]
Thank you for doing this! I'll review this PR this week.

**Comment by willkg** [comment]
I checked 10,000 Firefox crash reports to see how many retained the signature and which signatures changed. This is the set of signatures that changed.

Do any of these suggest changes we should make? Are there other kinds of crash reports we should be checking?

[1777954_signature_check.txt](https://github.com/mozilla-services/socorro/files/9136202/1777954_signature_check.txt)

**Comment by gabrielesvelto** [comment]
Alright, pushed a handful of fixes, we should be good

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 1]
```diff
@@ -103,12 +110,21 @@ NS_ProcessNextEvent
...
+_platform_memcmp
+_platform_memset
+_platform_strchr
+_platform_strlen
+_platform_strncmp
+_platform_strncpy
+_platform_strstr
 PR_WaitCondVar
+__psynch_cvwait
+__pthread_.*
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:126]:
```suggestion
__pthread_
```

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 2]
```diff
@@ -121,29 +137,33 @@ RtlReportCriticalFailure
...
+__stack_chk_fail
 std::alloc::rust_oom
 std::io::stdio::_eprint
 std::panicking::
 std::sys_common::backtrace::__rust_end_short_backtrace<T>
 std::__terminate
 std::terminate
-system@framework@.*\.jar@classes\.dex@0x
-___TERMINATING_DUE_TO_UNCAUGHT_EXCEPTION___
+__strchr.*
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:158]:
```suggestion
__strchr
```

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 3]
```diff
@@ -121,29 +137,33 @@ RtlReportCriticalFailure
...
+__strcmp.*
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:159]:
```suggestion
__strcmp
```

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 4]
```diff
@@ -121,29 +137,33 @@ RtlReportCriticalFailure
...
+__strcpy_chk
+__strlen.*
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:161]:
```suggestion
__strlen
```

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 5]
```diff
@@ -121,29 +137,33 @@ RtlReportCriticalFailure
...
+__strncmp.*
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:162]:
```suggestion
__strncmp
```

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 6]
```diff
@@ -121,29 +137,33 @@ RtlReportCriticalFailure
...
+__strncpy.*
+__strnlen.*
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:164]:
```suggestion
__strnlen
```

**Comment by gabrielesvelto** [comment]
Oh I see, I guess this applies to all the other entries I added too such as `__memcmp.*`, `__memcpy.*`, etc... right? I'll remove the wildcard from all of them.

**Comment by gabrielesvelto** [comment]
I've removed all the redundant trailing `.*` expressions in the lists

**Comment by willkg** [comment]
Thank you for doing this!

**Pull request merged**